MFVNeutralino/test/ntuple.py

```diff
@@ -5,7 +5,7 @@
 settings = NtupleSettings()
 settings.is_mc = True
 settings.is_miniaod = True
-
+settings.event_filter_require_vertex = True
```

Comment:
I'm trying to understand if this can just be part of the configs, rather than as a setting to specify here in ntuple.py... 

Historically it only mattered for the EventFilter, but now it also is relevant to the "vertex filter ntuple" step in NtupleCommon.py. Oh, but I think it's only used when EventFilter is turned on? Just trying to understand if the name is still appropriate... At the least, if it is only used when EventFilter is on, we should have a corresponding comment where this setting is configured.

Reply:
Yes, only if EventFilter is on, this event_filter_require_vertex with be True. 
https://github.com/DisplacedVertices/cmssw-usercode/blob/d2c0260e403737b9440bb2850242f40d7f7dd3b4/MFVNeutralino/python/NtupleCommon.py#L269

Actually, this is also an earlier place that this setting controls what's going on in EventFilter.py:
https://github.com/DisplacedVertices/cmssw-usercode/blob/clean_vertexing/MFVNeutralino/python/NtupleCommon.py#L86
so it is no longer treated as a black box before the modification. I would propose that to avoid any confusion we can remove this setting from histos.py and just hardcode in NtupleCommon.py. + add a comment that it must be the same as the default setting here:
https://github.com/DisplacedVertices/cmssw-usercode/blob/88b6cd4859a8c1c6eb4aa900f19e4352ac52bb59/MFVNeutralino/python/EventFilter.py#L10

I don't see anywhere in the code that this default setting would change so hardcode?

Reply:
Hardcode it here in the settings, I suppose? https://github.com/DisplacedVertices/cmssw-usercode/blob/d2c0260e403737b9440bb2850242f40d7f7dd3b4/MFVNeutralino/python/NtupleCommon.py#L99

That probably would work well! Having it in a common place will be much better than in ntuple.py, since we do use NtupleCommon and the event filter elsewhere in our code.

Reply:
I'm not sure about this part though:
>add a comment that it must be the same as the default setting here:

What comment would need to go where?

Reply:
Note that we will come back to this later. A few concerns involve (1) a few places with `event_filter_require_vertex` (such as in tracker_mapper.py) (2) the mode should change `event_filter_require_vertex`.